### PR TITLE
fix: Each child in a list should have a unique "key" prop

### PR DESCRIPTION
**Pull request by sarobolket**
## Description
Fix the warning `Each child in a list should have a unique "key" prop`.
In the Table component there is an error because `headerGroup` don't have an id.

### CODE REVIEW

File: src/nft/components/explore/Table.tsx [hunk 1]
```diff
@@ -148,8 +148,8 @@ export function Table<D extends Record<string, unknown>>({
     <GlowEffect>
       <table {...getTableProps()} className={styles.table}>
         <thead className={styles.thead}>
-          {headerGroups.map((headerGroup) => (
-            <tr {...headerGroup.getHeaderGroupProps()} key={headerGroup.id}>
+          {headerGroups.map((headerGroup, index) => (
+            <tr {...headerGroup.getHeaderGroupProps()} key={index}>
```
Comment (tinaszheng) [inline src/nft/components/explore/Table.tsx:152]: for arrays like `headerGroups` that can dynamically change in size - don't use `index` as a key because it's not guaranteed that the same instance will have the same index

can you investigate why `headerGroup.id` is not working instead?

**Review by tinaszheng — CHANGES_REQUESTED** [review]

**Comment by sarobolket** [comment]
@tinaszheng Are there any problems with this PR?

File: src/nft/components/explore/Table.tsx [hunk 2]
```diff
@@ -149,7 +149,8 @@ export function Table<D extends Record<string, unknown>>({
       <table {...getTableProps()} className={styles.table}>
         <thead className={styles.thead}>
           {headerGroups.map((headerGroup) => (
-            <tr {...headerGroup.getHeaderGroupProps()} key={headerGroup.id}>
+            // eslint-disable-next-line react/jsx-key
+            <tr {...headerGroup.getHeaderGroupProps()}>
```
Comment (cbachmeier) [inline src/nft/components/explore/Table.tsx:153]: the table row should be given a valid key instead of removing and disabling the error

File: src/nft/components/explore/Table.tsx [hunk 3]
```diff
@@ -246,7 +247,8 @@ function LoadingTable({ headerGroups, visibleColumns, ...props }: LoadingTablePr
       <table {...props} className={styles.table}>
         <thead className={styles.thead}>
           {headerGroups.map((headerGroup) => (
-            <tr {...headerGroup.getHeaderGroupProps()} key={headerGroup.id}>
+            // eslint-disable-next-line react/jsx-key
+            <tr {...headerGroup.getHeaderGroupProps()}>
```
Comment (cbachmeier) [inline src/nft/components/explore/Table.tsx:251]: the table row should be given a valid key instead of removing and disabling the error

File: src/nft/components/explore/CarouselCard.tsx [hunk 4]
```diff
@@ -291,7 +291,7 @@ const DEFAULT_TABLE_ELEMENTS = 12
 const LoadingTable = () => {
   return (
     <>
-      {[...Array(DEFAULT_TABLE_ELEMENTS)].map((index) => (
+      {Array.from({ length: DEFAULT_TABLE_ELEMENTS }, (_, i) => i + 1).map((_, index) => (
```
Comment (cbachmeier) [inline src/nft/components/explore/CarouselCard.tsx:294]: Was CarouselCard also getting a unique key warning? why was this added?

File: src/nft/components/explore/Table.tsx [hunk 5]
```diff
@@ -280,7 +282,7 @@ function LoadingTable({ headerGroups, visibleColumns, ...props }: LoadingTablePr
           ))}
         </thead>
         <tbody {...props}>
-          {[...Array(DEFAULT_TRENDING_TABLE_QUERY_AMOUNT)].map((_, index) => (
+          {Array.from({ length: DEFAULT_TRENDING_TABLE_QUERY_AMOUNT }, (_, i) => i + 1).map((_, index) => (
```
Comment (cbachmeier) [inline src/nft/components/explore/Table.tsx:285]: what advantage does this syntax have over the current?

File: src/nft/components/explore/Table.tsx (hunk 2 above)
Comment (sarobolket) [reply src/nft/components/explore/Table.tsx:153]: The getHeaderGroupProps function return a key prop

File: src/nft/components/explore/CarouselCard.tsx (hunk 4 above)
Comment (sarobolket) [reply src/nft/components/explore/CarouselCard.tsx:294]: Yes, CarouselCard also getting a unique key warning.
The `[...Array(DEFAULT_TABLE_ELEMENTS)]` code return an array of undefined, instead `Array.from({ length: DEFAULT_TABLE_ELEMENTS }, (_, i) => i + 1)` return an array of number element

File: src/nft/components/explore/Table.tsx (hunk 5 above)
Comment (sarobolket) [reply src/nft/components/explore/Table.tsx]: you are right, there are no advantages, We can use `[...Array(DEFAULT_TRENDING_TABLE_QUERY_AMOUNT)].map((_, index)`

**Comment by mattbspector** [comment]
closing this - no code review in a significant amount of time - appreciate the PR and if you think this is still relevant feel free to reopen.
